### DISCUSSION TITLE
Sharingan ephemeral testnet

### DISCUSSION THREAD

**Discussion by AbdelStark** [keep-starknet-strange/madara#553] [Announcements]
<a href="https://github.com/keep-starknet-strange/madara/blob/main/docs/sharingan-starter-pack.md">
<img src="https://img.shields.io/badge/Testnet%20starter%20pack-Guide-green?logo=github"
 height="50" />
</a>
<a href="https://starknet-madara.netlify.app/?rpc=wss%3A%2F%2Fsharingan.cartridge.gg/#/explorer">
<img src="https://img.shields.io/badge/Dev-Explorer-yellow"
 height="50" />
</a>

# Network information
- Release / tag: https://github.com/keep-starknet-strange/madara/releases/tag/v0.1.0-testnet-sharingan-alpha.4
- Docker image: `ghcr.io/keep-starknet-strange/madara:v0.1.0-testnet-sharingan-alpha.4` (https://github.com/keep-starknet-strange/madara/pkgs/container/madara)
- Bootnode: `/ip4/52.7.206.208/tcp/30333/p2p/12D3KooWJytWW4wqhG1xp9ckLb7B15KqDU24Q8HHo8VfwXmFe5ZE`
- Dev explorer: https://starknet-madara.netlify.app/?rpc=wss%3A%2F%2Fsharingan.madara.wtf/#/explorer

# Operators

| Maintainer | ID | Key alias | IP | Peer ID | RPC port |
|------------|----|-----------|----|---------|----------|
| Starkware  | 1  | alice     | `52.7.206.208` | `12D3KooWK26CZBpWtwMaVQ6nXHrrXBkdXXx6CrBAU2KgLTqDNL6o` | `9944` |
| Starkware  | 2  | bob     | `44.195.161.82` | `12D3KooWMyrW5SvZ1WpcMLi7QdjXRQ36mUBg6RoaoABtPNMGVqXr` | `9944` |
| Cartridge  | 3  | charlie     | `208.67.222.222` | `12D3KooWQe2ZtqiyC5CLJKZr9i9xTmEyiAikZcr5J18w3cG1dQAc` | `9933` |
| LambdaClass  | 4  | dave     | `65.108.65.148` | `12D3KooWL3H14Kw4GzuNZHrmNwPNNpEGpXveLN5NVgpH4zBfbUrK` | `9944` |
| Pragma  | 5  | eve     | `13.39.22.82` | `12D3KooWB6fTFDBiALwGyCW6iRkfn9NwortsBxxEQUfCArgkGWdK` | `9933` |
| Kakarot  | 6  | ferdie     | `52.50.242.182` | `12D3KooWHnQ8LC113DgB5cVVyx2mvTN7bBkm75zvzsndr2WhstEE` | `9933` |

## Run your sequencer

<a href="https://gist.github.com/glihm/dcb92a18efbefb374b21e28e9144712f">
<img src="https://img.shields.io/badge/Testnet%20starter%20pack-Guide-green?logo=github"
 height="20" />
</a>

## Join as a full node

<a href="https://gist.github.com/glihm/dcb92a18efbefb374b21e28e9144712f">
<img src="https://img.shields.io/badge/Testnet%20starter%20pack-Guide-green?logo=github"
 height="20" />
</a>

## Get your peer id

```bash
curl --header "Content-Type: application/json" \
  --request POST \
  --data '{
    "jsonrpc": "2.0",
    "method": "system_localPeerId",
    "params": [],
    "id": 1
}' \
 http://44.195.161.82:9933
```

Response should look like this:
```json
{
  "jsonrpc": "2.0",
  "result": "12D3KooWHocNfvLz6rgpb8wJsynSpMwkspkcRn6gmN5UiK1tTTeG",
  "id": 1
}
```

**Comment by ivpavici** [comment]
Hi! Is it possible to add some recommended hardware requirements to join the testnet here as well?
I dug this up from telegram:
3 EC2 instances (c5.xlarge ?) (created automatically), 4 dedicated CPU-s, 16 GB RAM

**Reply by AbdelStark** [reply]
very good point. @glihm can you please update the Sharingan starter pack ?
btw @ivpavici next version will be much more easy to join as a full node.
@d-roak created a shortcut command.
You will just have to run: `madara --testnet sharingan`

**Reply by glihm** [reply]
@abdelhamidbakhta if it's ok for you I am still monitoring my node and collect some data to give an overview of this in the starter pack.

@ivpavici excellent remark, I totally missed it in the guide.

Keep you posted. 🫡

**Reply by AbdelStark** [reply]
Wonderful @glihm thank you

**Reply by glihm** [reply]
PR submitted, @ivpavici if you are not planning to surcharge your node with stuff like indexing for instance, you can start with `t2.medium` (2 vCPU and 4 GB RAM) instance without any trouble!

Then, if you are planning to send 20+ transactions per seconds or requesting data like indexers do, then 4 vCPU and 8/16GB ram would be great for responsiveness.

But as a fullnode, requirements are much lower than sequencers participating to the network.
